internal/eval/eval.go

```diff
@@ -675,6 +675,105 @@ func evalJSONSET(args []string, store *dstore.Store) []byte {
 	return clientio.RespOK
 }
 
+// evalJSONARRAPPEND appends the value(s) provided in the args to the given array path
+// in the JSON object saved at key in arguments.
+// Args must contain atleast a key, path and value.
+// If the key does not exist or is expired, it returns response.RespNIL.
+// If the object at given path is not an array, it returns response.RespNIL.
+// Returns the new length of the array at path.
+func evalJSONARRAPPEND(args []string, store *dstore.Store) []byte {
+    if len(args) < 3 {
+        return diceerrors.NewErrArity("JSON.ARRAPPEND")
+    }
+
+    var key string = args[0]
+    var path string = args[1]
+
+	obj := store.Get(key)
+	if obj == nil {
+		return diceerrors.NewErrWithMessage("could not perform this operation on a key that doesn't exist")
+	}
+
+	errWithMessage := dstore.AssertTypeAndEncoding(obj.TypeEncoding, dstore.ObjTypeJSON, dstore.ObjEncodingJSON)
+	if errWithMessage != nil {
+		return errWithMessage
+	}
+
+	jsonData := obj.Value
+	_, err := sonic.Marshal(jsonData)
+	if err != nil {
+		return diceerrors.NewErrWithMessage("Existing key has wrong Dice type")
+	}
+
+	expr, err := jp.ParseString(path)
+	if err != nil {
+		return diceerrors.NewErrWithMessage("invalid JSONPath")
+	}
+
+	results := expr.Get(jsonData)
+    if(len(results) == 0){
+        return clientio.RespNIL
+    }
```

Comment:
thanks for your add this command. I have a few comments.
if the len(results) == 0. in redis. it should return a empty array
`

    127.0.0.1:6379> JSON.SET item:1 $ '{"name":"Noise-cancelling Bluetooth headphones","description":"Wireless Bluetooth headphones with noise-cancelling technology","connection":{"wireless":true,"type":"Bluetooth"},"price":99.98,"stock":25,"colors":["black","silver"]}'
    OK
    127.0.0.1:6379> JSON.ARRAPPEND item:1 $.aaa  '"a"'
    (empty array)

`

Reply:
sure will fix this. I still had doubts for the results array. Can one path return results with multiple values? At some places that path argument in the command could be a list and at some places its not. Will confirm this again from the docs.
<img width="403" alt="image" src="https://github.com/user-attachments/assets/56c4f37e-453f-467b-9b6c-c6e581ebd736">

Reply:
you could test it. the following is I tested in redis
`

      
    127.0.0.1:6379> JSON.SET user $ '{"age":13,"high":1.60,"pet":null,"language":["python","golang"],"flag":false,"partner":{"name":"tom","language":["rust"]}}'
    OK
    127.0.0.1:6379> JSON.GET user $.language
    "[[\"python\",\"golang\"]]"
    127.0.0.1:6379> JSON.GET user $..language
    "[[\"python\",\"golang\"],[\"rust\"]]"
    127.0.0.1:6379> JSON.ARRAPPEND user $..language '"java"'
    1) (integer) 3
    2) (integer) 2
    127.0.0.1:6379> JSON.GET user $..language
    "[[\"python\",\"golang\",\"java\"],[\"rust\",\"java\"]]"

`

Reply:
right, I will fix this

Reply:
> sure will fix this. I still had doubts for the results array. Can one path return results with multiple values? At some places I read that path argument in the command could be a list and at some places its not. Will confirm this again from the docs. <img alt="image" width="403" src="https://private-user-images.githubusercontent.com/34770136/366909469-56c4f37e-453f-467b-9b6c-c6e581ebd736.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxNTAzNjAsIm5iZiI6MTcyNjE1MDA2MCwicGF0aCI6Ii8zNDc3MDEzNi8zNjY5MDk0NjktNTZjNGYzN2UtNDUzZi00NjdiLTliNmMtYzZlNTgxZWJkNzM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDE0MDc0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxMzdjOWNiMGRkMTliY2ViYWZjNTdmZDRkYjJjZTQ1NmY0M2JhMmNiYjQ2OGMxMzBiZjdjZTIyNDExYTI3Y2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uC4in6BM7ogcGnCp2ex8h6hIFAcgMBfXpHoZUh_GzJ8">

I observer that in some json command. if the **path** is exist or not exist, will determine the return value if is an array,  in this issue, the redisJson contributor explained why they use array [https://github.com/RedisJSON/RedisJSON/issues/882](url)